mbed_flasher/main.py

```diff
@@ -222,61 +299,73 @@ def subcmd_flash_handler(self, args, logger):
-                                            if not device['platform_name'] in available_platforms:
-                                                available_platforms.append(device['platform_name'])
-                                        target_ids_to_flash.append(device['target_id'])
-                    else:
-                        for device in available:
-                            available_target_ids.append(device['target_id'])
-                            if device['target_id'] == args.tid or device['target_id'].startswith(args.tid):
-                                if not device['target_id'] in target_ids_to_flash:
-                                    if 'platform_name' in device:
-                                        if not device['platform_name'] in available_platforms:
-                                            available_platforms.append(device['platform_name'])
-                                    target_ids_to_flash.append(device['target_id'])
-                    if len(target_ids_to_flash) == 0:
-                        print("Could not find given target_id from attached devices")
-                        print("Available target_ids:")
-                        print(available_target_ids)
-                        return EXIT_CODE_COULD_NOT_MAP_DEVICE
-                    else:
-                        if len(available_platforms) > 1:
-                            if not args.platform_name:
-                                print("More than one platform detected for given target_id")
-                                print("Please specify the platform with -t <PLATFORM_NAME>")
-                                print("Found platforms:")
-                                print(available_platforms)
-                                return EXIT_CODE_PLATFORM_REQUIRED
-                        else:
-                            #print(target_ids_to_flash)
-                            retcode = flasher.flash(build=args.input, target_id=target_ids_to_flash, platform_name=available_platforms[0], method=args.method, no_reset=args.no_reset)
-                else:
-                    print("Could not find any connected device")
-                    return EXIT_CODE_DEVICES_MISSING
-            else:
-                retcode = flasher.flash(build=args.input, target_id='all', platform_name=args.platform_name, method=args.method, no_reset=args.no_reset)
-        else:
+
+        if not args.tid:
             print("Target_id is missing")
             return EXIT_CODE_NO_TARGET_ID
-        
+
+        if 'all' in args.tid:
+            retcode = flasher.flash(build=args.input, target_id='all',
+                                    platform_name=args.platform_name,
+                                    method=args.method, no_reset=args.no_reset)
+
+        if len(available) <= 0:
+            print("Could not find any connected device")
+            return EXIT_CODE_DEVICES_MISSING
+
+        if isinstance(args.tid, list):
```

Comment:
I refactored this part in [this commit](https://github.com/ARMmbed/mbed-flasher/pull/55/commits/c68b0d92e0f107d1d30383fdbd25f812d646ad05)
could you please check if it's correct and what else can be improved.